src/rest/EndpointCaller.ts

```diff
@@ -206,6 +211,27 @@ export class EndpointCaller {
     this.logger = new Logger(this);
   }
 
+  public static convertJsonToQueryString(json: { [key: string]: any }): string[] {
+    Assert.exists(json);
+
+    const result: string[] = [];
+    _.each(json, (value, key) => {
+      if (value != null) {
+        if (_.isObject(value)) {
```

Comment:
There is some repetition in both branches of the `if/else`. It would be better to use the `isObject`  condition to determine the right value, and then continue. e.g.

```
const stringValue = _.isObject(value) ? JSON.stringify(value) : value.toString();
const formUrlEncoded = key + '=' + Utils.safeEncodeURIComponent(stringValue);
result.push(formUrlEncoded);
```